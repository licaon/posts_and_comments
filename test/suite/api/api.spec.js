import fetchData from 'src/api';

describe('Api', () => {
  it(`should execute fetchData correctly and default to GET method`, () => {
    fetch.mockResponse(JSON.stringify({
      response: 'response',
    }));

    return fetchData({ url: 'url' }).then((data) => {
      expect(fetch).toHaveBeenCalledWith(
        'url',
        {
          body: {},
          headers:
            {
              Accept: 'application/json',
              'Content-Type': 'application/json',
            },
          method: 'GET',
        },
      );
      expect(data).toMatchSnapshot();
    });
  });
  it(`should execute fetchData correctly with the right arguments`, () => {
    fetch.mockResponse(JSON.stringify({
      response: 'response',
    }));

    return fetchData({
      url: 'url',
      body: {
        content: 'content',
      },
      method: 'POST',
    }).then((data) => {
      expect(fetch).toHaveBeenCalledWith(
        'url',
        {
          body: {
            content: 'content',
          },
          headers:
            {
              Accept: 'application/json',
              'Content-Type': 'application/json',
            },
          method: 'POST',
        },
      );
      expect(data).toMatchSnapshot();
    });
  });
});
