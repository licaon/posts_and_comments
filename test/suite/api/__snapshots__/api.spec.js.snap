// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api should execute fetchData correctly and default to GET method 1`] = `
Object {
  "response": "response",
}
`;

exports[`Api should execute fetchData correctly with the right arguments 1`] = `
Object {
  "response": "response",
}
`;
